jmxtrans-output/jmxtrans-output-elastic/src/main/java/com/googlecode/jmxtrans/model/output/elastic/ElasticWriter.java

```diff
@@ -0,0 +1,237 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static com.fasterxml.jackson.core.JsonEncoding.UTF8;
+import static com.googlecode.jmxtrans.model.PropertyResolver.resolveProps;
+import static com.googlecode.jmxtrans.model.naming.KeyUtils.getKeyString;
+import static com.googlecode.jmxtrans.util.NumberUtils.isNumeric;
+
+/**
+ * Feed data directly into elastic.
+ *
+ * @author Peter Paul Bakker - pp@stokpop.nl
+ */
+
+@NotThreadSafe
+public class ElasticWriter extends BaseOutputWriter {
+	
+	private static final Logger log = LoggerFactory.getLogger(ElasticWriter.class);
+	
+	private static final String DEFAULT_ROOT_PREFIX = "jmxtrans";
+	private static final Object CREATE_MAPPING_LOCK = new Object();
+
+	private final JsonFactory jsonFactory;
+
+	private JestClient jestClient = null;
+
+	private final String rootPrefix;
+	private final String connectionUrl;
+	private static final String TYPE_NAME = "jmx-entry";
+	private String indexName;
+
+	@JsonCreator
+	public ElasticWriter(
+			@JsonProperty("typeNames") ImmutableList<String> typeNames,
+			@JsonProperty("booleanAsNumber") boolean booleanAsNumber,
+			@JsonProperty("rootPrefix") String rootPrefix,
+			@JsonProperty("debug") Boolean debugEnabled,
+			@JsonProperty("connectionUrl") String connectionUrl,
+			@JsonProperty("settings") Map<String, Object> settings) {
+
+		super(typeNames, booleanAsNumber, debugEnabled, settings);
+
+		this.rootPrefix = resolveProps(
+				firstNonNull(
+						rootPrefix,
+						(String) getSettings().get("rootPrefix"),
+						DEFAULT_ROOT_PREFIX));		
+
+		this.jsonFactory = new JsonFactory();
+		this.connectionUrl = connectionUrl;
+		this.indexName = this.rootPrefix + "_jmx-entries";
+
+	}
+
+	@Override
+	protected void internalWrite(Server server, Query query, ImmutableList<Result> results) throws Exception {
+		List<String> typeNames = this.getTypeNames();
+
+		for (Result result : results) {
+			log.debug("Query result: [{}]", result);
+			Map<String, Object> resultValues = result.getValues();
+			for (Entry<String, Object> values : resultValues.entrySet()) {
+				Object value = values.getValue();
+				if (isNumeric(value)) {
+					String message = createJsonMessage(server, query, typeNames, result, values, value);
+					log.debug("Insert into Elastic: Index: [{}] Type: [{}] Message: [{}]", indexName, TYPE_NAME, message);
+					Index index = new Index.Builder(message).index(indexName).type(TYPE_NAME).build();
+					jestClient.execute(index);
+				} else {
+					log.warn("Unable to submit non-numeric value to Elastic: [{}] from result [{}]", value, result);
+				}
+			}
+		}
+	}
+
+	private String createJsonMessage(Server server, Query query, List<String> typeNames, Result result, Entry<String, Object> values, Object value) throws IOException {
+
+		String keyString = getKeyString(server, query, result, values, typeNames, this.rootPrefix);
+
+		String alias;
```

Comment:
You should extract a `computeAlias()` method from this block (possibly with a better name).